src/onegov/event/collections/occurrences.py

```diff
@@ -455,28 +453,20 @@ def maybe_sorted(values: 'Iterable[str]') -> list[str]:
     def used_tags(self) -> set[str]:
         """ Returns a list of all the tags used by all future occurrences.
 
-        This could be solve possibly more effienciently with the skey function
+        This could be solved possibly more efficiently with the skey function
         currently not supported by SQLAlchemy (e.g.
         ``select distinct(skeys(tags))``), see
         http://stackoverflow.com/q/12015942/3690178
 
         """
-        base = self.session.query(
-            # FIXME: Shouldn't we just directly query the two attributes
-            #        below rather than use with_entities?
-            Occurrence._tags.keys()  # type:ignore[attr-defined]
-        ).with_entities(
-            sqlalchemy.func.skeys(Occurrence._tags).label('keys'),
-            Occurrence.end
-        )
-        base = base.filter(func.DATE(Occurrence.end) >= date.today())
 
-        query = sqlalchemy.select(
-            [sqlalchemy.func.array_agg(sqlalchemy.column('keys'))],
-            distinct=True
-        ).select_from(base.subquery())
-        keys = self.session.execute(query).scalar()
-        return set(keys) if keys else set()
+        tags = set()
+        future_occurrences = (self.session.query(Occurrence)
+                              .filter(func.DATE(Occurrence.end)
+                                      >= date.today()).all())
+        for occurrence in future_occurrences:
+            tags.update(occurrence.tags)
+        return tags
```

Comment:
Looking at this again, I think the following would be a lot more sane, I don't think there is much of a point to pack it into an array at Postgres' end, since we just repack it into a set anyways.

```python
        query = self.session.query(
            sqlalchemy.func.skeys(Occurrence._tags)
        ).filter(func.DATE(Occurrence.end) >= date.today())
        return {key for key, in query.distinct()}
```

You can also remove most of the docstring, since we are already using the `skeys` function.